File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        int cnt = 0;
        bitset<10001>vis;
        for(auto it : banned)vis[it] = 1;
        for(int i = 1 ; i <= n ;i++)// n
            if(!vis[i] && i <= maxSum)cnt++,maxSum-=i;
        return cnt;
        
    }
};

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.js
var maxCount = function(banned, n, maxSum) {
    const bannedSet = new Set(banned);  // Convert banned array to a set
    let count = 0;
    
    for (let i = 1; i <= n; i++) {
        if (!bannedSet.has(i) && i <= maxSum) {
            count++;
            maxSum -= i;
        }
    }
    
    return count;
};
